autoload/ack.vim

```diff
@@ -64,26 +64,59 @@ function! ack#show_results()
   execute s:handler
 endfunction
 
+function! ack#preview()
+  let idqf = line('.') - 1
+  let pos = getqflist()[idqf]
+  let lnn = pos.lnum
+
+  if lnn < 1
+    let lnn = 1
+  endif
+
+  let fname = pos.bufnr
+
+  if type(fname) == type(0)
+    let fname = bufname(fname)
+  endif
+  if fname == '%'
+    let fname = bufname('%')
+  endif
+
+  if fname == ''
+    return
+  endif
+
+  exec 'topleft pedit ' . '+' . lnn . ' ' . fname
```

Comment:
Minor, but this can have less syntax noise as `exec 'topleft pedit +' . lnn fname`.